ably/realtime_channel_spec_test.go

```diff
@@ -711,3 +711,87 @@ func TestRealtimeChannel_RTL13_HandleDetached(t *testing.T) {
 		ablytest.Instantly.NoRecv(t, nil, out, t.Fatalf)
 	})
 }
+
+func TestRealtimeChannel_RTL14_HandleChannelError(t *testing.T) {
+	t.Parallel()
+
+	const channelRetryTimeout = 123 * time.Millisecond
+
+	setup := func(t *testing.T) (
+		in, out chan *proto.ProtocolMessage,
+		c *ably.Realtime,
+		channel *ably.RealtimeChannel,
+		stateChanges ably.ChannelStateChanges,
+		afterCalls chan ablytest.AfterCall,
+	) {
+		in = make(chan *proto.ProtocolMessage, 1)
+		out = make(chan *proto.ProtocolMessage, 16)
+		afterCalls = make(chan ablytest.AfterCall, 1)
+		now, after := ablytest.TimeFuncs(afterCalls)
+
+		c, _ = ably.NewRealtime(
+			ably.WithToken("fake:token"),
+			ably.WithAutoConnect(false),
+			ably.WithNow(now),
+			ably.WithAfter(after),
+			ably.WithChannelRetryTimeout(channelRetryTimeout),
+			ably.WithDial(ablytest.MessagePipe(in, out)),
+		)
+
+		in <- &proto.ProtocolMessage{
+			Action:            proto.ActionConnected,
+			ConnectionID:      "connection-id",
+			ConnectionDetails: &proto.ConnectionDetails{},
+		}
+
+		err := ablytest.Wait(ablytest.ConnWaiter(c, c.Connect, ably.ConnectionEventConnected), nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		channel = c.Channels.Get("test")
+
+		ctx, cancel := context.WithCancel(context.Background())
+		go channel.Attach(ctx)
```

Comment:
As explained [here](https://github.com/ably/ably-go/pull/288#discussion_r587259552), please don't use plain `go` in tests.

Reply:
Actually, it's being used in another setup script too
 https://github.com/ably/ably-go/blob/5b39ec35624c26f76c9bf036d4fe8b1b2d334ce8/ably/realtime_channel_spec_test.go#L512

Reply:
Should it be changed there too?

Reply:
@sacOO7 Yes, certainly. It doesn't need to be part of this PR though.